### PR TITLE
Governance.ContentTypeEnforceRetention & Core.UserProfiles.Sync

### PR DESCRIPTION
**Pull request by SharePointRadi**
This pull has 2 projects in the Solution folder:
Governance.ContentTypeEnforceRetention - a console app to crawl content in a given site collection and perform a retention action if the last modified date is too old. (See documentation for actual CSOM limitation and business problem).
Core.UserProfile.Sync - A custom sync implementation authenticating to Azure AD and SPO UPA and syncing custom fields that are not part of the Office 365 UP sync fields.

### CODE REVIEW

**Comment by erwinvanhunen** [comment]
These are two -really- great additions Radi! Thanks! In the readme.md for both projects you refer to an image explaining AppRegNew.aspx and AppInv.aspx, in those images the forms are partly filled in to show an app called SiteUsageReport. That might cause confusion?

File: Solutions/Core.UserProfiles.Sync/Core.UserProfiles.Sync/App.config [hunk 1]
```diff
@@ -0,0 +1,40 @@
+﻿<?xml version="1.0" encoding="utf-8"?>
+<configuration>
+  <appSettings>
+    <!--Used in SharePoint and in Azure AD-->
+    <add key="ClientId" value="GUID HERE />
```
Comment (erwinvanhunen) [inline Solutions/Core.UserProfiles.Sync/Core.UserProfiles.Sync/App.config]: You're not closing the value tag here, causing the project not to compile

**Comment by VesaJuvonen** [comment]
Great stuff. With those small adjustments these are definitely ready to go and highly valuable. Great stuff. Thanks.

**Comment by SharePointRadi** [comment]
I have corrected the missing quotes in app.config and updated the images in the readme.md :)
This one was a "team effort". We (me and my colleagues) still need to understand GitHub better.

**Comment by erwinvanhunen** [comment]
Thanks! All merged :-)

**Pull request merged**